bgpd/bgp_network.c

```diff
@@ -903,10 +903,9 @@ int bgp_getsockname(struct peer *peer)
 	}
 
 	peer->su_local = sockunion_getsockname(peer->connection->fd);
-	if (!peer->su_local)
-		return -1;
 	peer->su_remote = sockunion_getpeername(peer->connection->fd);
-	if (!peer->su_remote)
+
+	if (!peer->su_local || !peer->su_remote)
```

Comment:
Actually, these checks are not needed at all, because `bgp_zebra_nexthop_set()` is technically the same, right?